hazelcast/include/hazelcast/client/serialization/serialization.h

```diff
@@ -812,6 +813,11 @@ class HAZELCAST_API object_data_input
       std::is_base_of<custom_serializer, hz_serializer<T>>::value,
       boost::optional<T>>::type inline read_object(int32_t type_id);
 
+    template<typename T>
```

Comment:
inline missing for read_object and write_object of generic_record type.

Reply:
`inline` is not needed here but somehow it is used. I don't think it is correct to do so. I think others should be removed as well.